logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/ProteusClientProvider.kt

```diff
@@ -0,0 +1,38 @@
+package com.wire.kalium.logic.feature
+
+import com.wire.kalium.cryptography.ProteusClient
+import com.wire.kalium.cryptography.ProteusClientImpl
+import com.wire.kalium.cryptography.exceptions.ProteusException
+
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlin.coroutines.cancellation.CancellationException
+
+interface ProteusClientProvider {
+    suspend fun clear()
+    suspend fun getOrCreate(): ProteusClient
```

Comment:
To avoid creating new files unintentionally, we can add another function
```kotlin
    suspend fun getOrError(): Either<CoreFailure, ProteusClient> {
          // check if the files exists or return MissingClientRegistration
}
```
because the creating new files should only happen during register client

Reply:
@saleniuk you added `getOrError()` but aren't using it. As I understood @MohamadJaara you use it everywhere and only use `getOrCreate()` when registering a new client.

Reply:
@typfel sure, my bad, now it's correct